src/etl/sds/etl_state_lock_enforcer/steps.py

```diff
@@ -36,58 +36,65 @@ def _process_sqs_message(data, cache: Cache):
     message = data[StepChain.INIT]
     body = json_loads(message["body"])
     state_machine_name = body["name"]
+    manual_retry_state = body["manual_retry"]
+    cache["manual_retry_state"] = manual_retry_state
 
     # Validate and create StateMachineInput instance
     state_machine_input = StateMachineInput(**body)
 
-    return state_machine_input, state_machine_name
+    return state_machine_input, state_machine_name, manual_retry_state
 
 
 def _check_etl_lock(data, cache: Cache):
-    _, state_machine_name = data[_process_sqs_message]
-
+    _, state_machine_name, manual_retry_state = data[_process_sqs_message]
     s3_client = cache["s3_client"]
     etl_bucket = cache["etl_bucket"]
 
-    if etl_state_lock_doesnt_exist_in_s3(s3_client=s3_client, bucket=etl_bucket):
-        # Aquire state lock
-        return s3_client.put_object(
-            Bucket=cache["etl_bucket"],
-            Key=ETL_STATE_LOCK,
-            Body=state_machine_name,
-        )
+    if not manual_retry_state:
+        if etl_state_lock_doesnt_exist_in_s3(s3_client=s3_client, bucket=etl_bucket):
+            # Aquire state lock
+            return s3_client.put_object(
+                Bucket=cache["etl_bucket"],
+                Key=ETL_STATE_LOCK,
+                Body=state_machine_name,
+            )
 
-    else:
-        raise StateLockExistsError("ETL state lock already exists.")
+        else:
+            raise StateLockExistsError("ETL state lock already exists.")
 
 
 def _validate_state_keys_are_empty(data, cache: Cache):
-    return validate_state_keys_are_empty(
-        s3_client=cache["s3_client"], bucket=cache["etl_bucket"]
-    )
+    _, _, manual_retry_state = data[_process_sqs_message]
```

Comment:
If you cached `manual_retry_state` in `process_sqs_message` step can you call it via that instead? Or keep calling this way and then the cache isn't needed?